substrate/client/authority-discovery/src/worker/addr_cache.rs

```diff
@@ -35,11 +45,70 @@ pub(super) struct AddrCache {
 	peer_id_to_authority_ids: HashMap<PeerId, HashSet<AuthorityId>>,
 }
 
+impl std::fmt::Debug for AddrCache {
```

Comment:
Better if I derive `Debug` on `Multiaddr` and `PeerId` and then we can derive `Debug` for `AddrCache`....